gtdynamics/universal_robot/sdf.cpp

```diff
@@ -106,13 +105,34 @@ LinkSharedPtr LinkFromSdf(uint8_t id, const sdf::Link &sdf_link) {
   const auto &I = sdf_link.Inertial().Moi();
   inertia << I(0, 0), I(0, 1), I(0, 2), I(1, 0), I(1, 1), I(1, 2), I(2, 0),
       I(2, 1), I(2, 2);
-  const auto wTl = Pose3FromIgnition(sdf_link.Pose());
+
+  /// Call SemanticPose::Resolve so the pose is resolved to the correct frame
+  /// http://sdformat.org/tutorials?tut=pose_frame_semantics&ver=1.7&cat=specification&
+  // Get non-const Pose3d
+  auto jTl = sdf_link.RawPose();
+  // Update from joint frame to base frame in-place.
+  // Base frame is denoted by "".
+  auto errors = sdf_link.SemanticPose().Resolve(jTl, "");
+  // If any errors in the resolution, throw an exception.
+  if (errors.size() > 0) {
+    throw std::runtime_error(errors[0].Message());
+  }
+  // Pose is updated from joint frame to world frame.
+  const auto wTl = Pose3FromIgnition(jTl);
```

Comment:
You say `w` but I say `b`, right? Or does "Pose3FromIgnition" magically change it from base frame to world frame?

Reply:
It should be world frame throughout. The comment before is incorrect.

Reply:
Should be `// World frame is denoted by "".`

Reply:
Wait, I'm wrong. Let me look at this closer.

Reply:
Yeah it should be `bTl`. Nice catch.